File: haskell/p16.hs
-- P16 (**) Drop every N'th element from a list.

myDrop :: [x] -> Int -> [x]
myDrop [] _ = []
-- Take n - 1 elements, then skip one.
myDrop x n  = take (n - 1) x ++ myDrop (drop n x) n

File: ben/prolog/P01-P28-lists/p16.pl
% Drop every N'th element from a list.

drop(L, Nb, R) :- drop(L, Nb, R, Nb). 

drop([], _, [], _).
drop([_|Xs], 1, R, Nb) :-
    drop(Xs, Nb, R, Nb).
drop([X|Xs], N0, [X|R], Nb) :-
    N0 > 1,
    N1 is N0 - 1,
    drop(Xs, N1, R, Nb).
